crates/sui-core/src/consensus_adapter.rs

```diff
@@ -371,6 +369,7 @@ impl ConsensusAdapter {
         );
         for transaction in recovered {
             if transaction.is_end_of_publish() {
+                epoch_store.set_own_end_of_publish_sent(true);
```

Comment:
Any potential issue with setting this to true before confirming the submit was successful?